src/language-handlebars/printer-glimmer.js

```diff
@@ -267,9 +267,21 @@ function print(path, options, print) {
         text = "\\" + text;
       }
 
+      const trimmedValue = text
+        .replace(/^\s+/g, leadingSpace)
+        .replace(/\s+$/, trailingSpace);
+
+      // Glimmer parser treats attribute values with mustaches differently
+      // they're a ConcatStatement rather than a single TextNode
+      // if the attribute value is a plain string, we only trim it and avoid inserting hardlines
+      // since they're not rendered correctly within attributes
+      if (isParentOfSomeType(path, ["AttrNode"])) {
```

Comment:
This conditionnal will match `<htmlElement attribute="textNode">`'s TextNode but not `<htmlElement attribute="textNodeInConcatStatement {{mustache}}">`'s.

I would suggest to use instead:
```suggestion
      if (path.stack.includes("attributes")) {
```
to match both.

Reply:
If you apply this suggestion, you'll notice that you can return `return concat([trimmedValue]);` from line 234 and get rid of the big `else` statement.

Doing that, we'd split the TextNode printer in two: one for `TextNode`s nested in attributes, one for other `TextNode`s. We'd be able later to print differently attributes whether they be `class` / `style` / `srcset` / other.

Reply:
> This conditionnal will match <htmlElement attribute="textNode">'s TextNode but not <htmlElement attribute="textNodeInConcatStatement {{mustache}}">'s.

Correct and that's the whole point — presence of `MustacheStatement` makes it `ConcatStatement` which is handled properly.

Reply:
It is handled correctly in the sense it doesn’t have the bug. But a TextNode in a ConcatStatement should probably be formatted no differently than a simple TextNode (ie. trimmed / not trimmed)?

Reply:
@dcyriller just tried your suggestion and remembered why I didn't do that. It breaks mustache handling in attributes:

```diff
    <div
    -   class='a-first-class
    -   {{if this.optionOne 'optionOne'}}
    -
    -    {{if this.optionTwo 'optionTwo'}}
    -
    -    {{if this.optionThree 'optionThree'}}
    -
    -    {{if this.optionFour 'optionFour'}}
    -
    -    {{if this.optionFive 'optionFive'}}
    -
    -    {{this.class}}'
    +   class='a-first-class {{if this.optionOne 'optionOne'}} {{
    +     if this.optionTwo 'optionTwo'
    +   }} {{if this.optionThree 'optionThree'}} {{
    +     if this.optionFour 'optionFour'
    +   }} {{if this.optionFive 'optionFive'}} {{this.class}}'
        ...attributes
      >
      </div>
```

Reply:
This ☝️ would actually be an improvement - if applied to only class names.

Reply:
Yeah, this is something I wanted to discuss with you too. In https://github.com/prettier/prettier/issues/6207 they ask to format classes with newlines (but not extra newlines) whereas I feel like ☝️ option is not bad. We could consider that ticket an improvement or bring it up for more discussion.